cmd/adwatchd/integration_tests/adwatchd_service_test.go

```diff
@@ -97,15 +85,53 @@ func TestServiceStateChange(t *testing.T) {
 					return
 				}
 				require.NoError(t, err, fmt.Sprintf("%s failed but shouldn't", state))
-
-				if tc.wantStopped {
-					out := getStatus(t, app)
-					require.Contains(t, out, "stopped", "Service should be stopped")
-				}
 			}
 		})
 	}
 }
+
+func TestServiceStartWithABadDir(t *testing.T) {
+	// This should error on Windows but doesn't with systemd because of the auto-restart policy
+	if runtime.GOOS != "windows" {
+		t.Skip()
+	}
+
+	var err error
+
+	watchDir := t.TempDir()
+	configPath := generateConfig(t, -1, watchDir)
+
+	app := commands.New(commands.WithServiceName("adwatchd-test-baddir"))
+
+	installService(t, configPath, app)
+	time.Sleep(time.Second)
+
+	// Begin with a stopped state
+	changeAppArgs(t, app, "", "service", "stop")
+	err = app.Run()
+	require.NoError(t, err, "Setup: Stopping the service failed but shouldn't")
+
+	// Remove watched dir to make the service not start
+	os.RemoveAll(watchDir)
+
+	// The test is flaky on Windows and returns "context deadline exceeded" from
+	// time to time, so we retry a few times
+	tries := 0
+	for tries < 5 {
+		changeAppArgs(t, app, "", "service", "start")
+		err = app.Run()
+		if err == nil || !errors.Is(err, context.DeadlineExceeded) {
+			break
+		}
+		tries++
+	}
+	time.Sleep(time.Second)
```

Comment:
What is this sleep doing? Should it be for getStatus, and thus, after require.NoError?